ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts/eos_designs_facts.py

```diff
@@ -0,0 +1,1201 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+import ipaddress
+from functools import cached_property
+from hashlib import sha256
+from ansible_collections.arista.avd.plugins.module_utils.utils import get, default, template_var
+
+
+class EosDesignsFactsError(Exception):
+    def __init__(self, message="An Error has occured in EosDesignsFacts"):
+        self.message = message
+        super().__init__(self.message)
+
+
+class EosDesignsFactsMissingVariableError(EosDesignsFactsError):
+    pass
+
+
+class EosDesignsFacts:
+    _hostvars = {}
```

Comment:
just a quick question - using mutable objects like list and dict as class variables can have surprising effects in python (cf https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables) is this the intention here?